include/LightGBM/dataset.h

```diff
@@ -682,6 +697,34 @@ class Dataset {
 
   void AddFeaturesFrom(Dataset* other);
 
+  /*! \brief Get has_raw_ */
+  inline bool has_raw() const { return has_raw_; }
+
+  /*! \brief Set has_raw_ */
+  inline void SetHasRaw(bool has_raw) { has_raw_ = has_raw; }
+
+  /*! \brief Resize raw_data_, use current number of features */
+  inline void ResizeRaw(int num_rows) {
+    if (static_cast<int>(raw_data_.size()) > num_numeric_features_) {
+      raw_data_.resize(num_numeric_features_);
+    }
+    for (size_t i = 0; i < raw_data_.size(); ++i) {
+      raw_data_[i].resize(num_rows);
+    }
+    int curr_size = raw_data_.size();
+    for (int i = curr_size; i < num_features_; ++i) {
```

Comment:
Even when the number of features and number of numerical features are not changed, this will always push a new vector in to raw_data_, as long as there is a numerical feature with inner feature index >= cur_size.